website/content/docs/security/acl/tokens/create/create-a-mesh-gateway-token.mdx

```diff
@@ -0,0 +1,364 @@
+  Learn how to create ACL tokens that your mesh gateway can present to Consul servers so that they can register with the Consul catalog.
+---
+
+# Create a mesh gateway token
+
+This topic describes how to create a token to enable a mesh gateway to register.
+
+## Introduction
+
+Gateways must present a token linked to policies that grant the appropriate set of permissions. For a mesh gateway to be discoverable and to route to other services in a mesh, the token must be linked to policies that grant the following permissions:
+
+* `mesh:write` to obtain leaf certificates for terminating TLS connections
+* `peering:read` for Consul cluster peering. If you are not using Consul cluster peering, then you can omit the `peering:read` permission.
+* `service:write` to allow the mesh gateway to register into the catalog
+* `service:read` for all services and `node:read` for all nodes in order to discover and route to services
+* `agent:read` to enable the `consul connect envoy` CLI command to automatically discover gRPC settings from the Consul agent. If this command is not used to start the gateway or if the Consul agent uses the default gRPC settings, then you can omit the `agent:read` permission.
+
+@include 'create-token-requirements.mdx'
+
+## Create a token linked to a policy
+
+To create a token for the mesh gateway, you must define a policy, register the policy with Consul, and link the policy to a token.
+
+### Define a custom policy
+
+You can send custom policy definitions as command line or API arguments or define them in an external HCL or JSON file. The following example policies are externally defined policies. Refer to [ACL Rules](/consul/docs/security/acl/acl-rules) for details about all of the rules you can use in your policies.
+
+#### Consul OSS
+
+The following example policy grants the mesh gateway the appropriate permissions for the mesh gateway to register as a service named `mesh-gateway`.
+
+<CodeTabs>
+
+```hcl
+mesh = "write"
+peering = "read"
+service "mesh-gateway" {
+   policy = "write"
+}
+service_prefix "" {
+   policy = "read"
+}
+node_prefix "" {
+   policy = "read"
+}
+agent_prefix "" {
+   policy = "read"
+}
+```
+
+```json
+{
+  "mesh": "write",
+  "peering": "read",
+  "service": {
+    "mesh-gateway": [{
+      "policy": "write"
+    }]
+  },
+  "service_prefix": {
+    "": [{
+      "policy": "read"
+    }]
+  },
+  "node_prefix": {
+    "": [{
+      "policy": "read"
+    }]
+  },
+  "agent_prefix": {
+    "": [{
+      "policy": "read"
+    }]
+  }
+}
+```
+
+</CodeTabs>
+
+
+#### Consul Enterprise using `default` partition
+
+You can specify an admin partition and namespace when using Consul Enterprise. Mesh gateways must register into the `default` namespace. The following policy allows a mesh gateway to register as a service named `mesh-gateway` in the default partition.
+
+<CodeTabs>
+
+```hcl
+mesh = "write"
+peering = "read"
+partition_prefix "" {
+  peering = "read"
+}
+namespace "default" {
```

Comment:
Does this run into the mixing of scope levels issue we recently discovered (e.g., one rule at partition level, but another at the namespace level without an explicit partition scope)?

Reply:
Yes. Should be fixed.